Sources/ClientRuntime/Networking/Http/Middlewares/DeserializeMiddleware.swift

```diff
@@ -27,7 +27,11 @@ public struct DeserializeMiddleware<OperationStackOutput>: Middleware {
 
             let response = try await next.handle(context: context, input: input) // call handler to get http response
             var copiedResponse = response
-            if (200..<300).contains(response.httpResponse.statusCode.rawValue) {
+            while (100..<200).contains(await response.httpResponse.statusCode.rawValue) {
+                // Wait until http response status code gets set to either success or error
+                try await Task.sleep(nanoseconds: 10)
```

Comment:
Is this the change that was required for tests to pass on Linux?

Reply:
Yeah, in Linux the integration tests failed sporadically because while the status code is 102 (which indicates to client that a full request has been received and the server is working on it), the code progresses to deserialization which caused it to throw errors. By making sure the status code is one of the following: success, fail, error, this sporadic failure was fixed.

Reply:
Looping is not the preferred way to handle a case like this, because it will result in extra CPU being consumed and make the rest of the app less responsive, even when an async wait is performed as you have here (10ns is way too short BTW.)

A better way is to provide a method on `HttpResponse` that will suspend execution until the status code changes to the desired value.

You would do that with a callback like this (code is over-simplified and maybe not syntactically correct, but gives you the feel for it):
```swift
    var continuation = CheckedContinuation<Void>?

    func waitForFinalStatus() async {
        return await withCheckedContinuation { continuation in
            self.continuation = continuation
        }
    }

    func setStatusCode(_ statusCode: Int) {
        self.statusCode = statusCode
        if code >= 200 {
            continuation?.resume()
        }
    }
```

Reply:
(I fixed a similar polling solution that we used on data streams last summer in this manner)